programs/stake/src/stake_instruction.rs

```diff
@@ -64,14 +64,29 @@ declare_process_instruction!(Entrypoint, DEFAULT_COMPUTE_UNITS, |invoke_context|
         Ok(me)
     };
 
+    // The EpochRewards sysvar only exists after the
+    // enable_partitioned_epoch_reward feature is activated. If it exists, check
+    // the `active` field
+    let epoch_rewards_active = invoke_context
+        .get_sysvar_cache()
+        .get_epoch_rewards()
+        .map(|epoch_rewards| epoch_rewards.active)
+        .unwrap_or(false);
```

Comment:
cc @2501babe on this change, since the bpf version of the stake program will need to fetch the epoch rewards using the new syscall